### PR TITLE
Initial Post Restore compiler options processing

### PR DESCRIPTION
**Pull request by dsouzai**
This PR adds the initial framework and infrastructure needed to process options post restore.

* Ensure `_postRestoreOptionSets` is initialized
* Add framework to process JIT, JITServer, and externally documented options
* Add framework to compensate options that follow Policy 3 described in https://github.com/eclipse-openj9/openj9/issues/16714

~depends on https://github.com/eclipse/omr/pull/6886~ (merged)
depends on https://github.com/eclipse/omr/pull/6896

### CODE REVIEW

**Comment by dsouzai** [comment]
@mpirvu or @ymanton could you please review? @vijaysun-omr fyi.

**Review by mpirvu — COMMENTED** [review]
Looks ok overall. This is more like a skeleton since I see many TODOs throughout the code.

File: runtime/compiler/control/J9Options.cpp [hunk 1]
```diff
@@ -3285,6 +3285,7 @@ J9::Options::packOptions(const TR::Options *origOptions)
    uint8_t *curPos = ((uint8_t *)options) + sizeof(TR::Options);
 
    options->_optionSets = NULL;
+   options->_postRestoreOptionSets = NULL;
```
[the remark this answers is not in the document]
Comment (dsouzai) [reply runtime/compiler/control/J9Options.cpp:3288]: I didn't see `_optionSets` in `unpackOptions`, so I guess we don't need to do anything about it?

File: runtime/compiler/control/OptionsPostRestore.hpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+   private:
+   void processCompilerOptions();
+   };
+
+}
+
+#endif // J9VM_OPT_CRIU_SUPPORT
+
+#endif // OPTIONS_POST_RESTORE_HPP
```
[the remark this answers is not in the document]
Comment (dsouzai) [reply runtime/compiler/control/OptionsPostRestore.hpp]: Done in force push; it doesn't seem to show the new line change, but it also isn't complaining about it.

**Comment by mpirvu** [comment]
jenkins test sanity all jdk17 depends eclipse/omr#6896

**Comment by dsouzai** [comment]
@mpirvu the OMR changes have made it to the `openj9` branch.

**Review by mpirvu — APPROVED** [review]

**Pull request merged**